server/lib/stylesheet-manager.js

```diff
@@ -3,11 +3,51 @@ const denodeify = require('denodeify');
 const fs = require('fs');
 const readFile = denodeify(fs.readFile);
 const chokidar = require('chokidar');
+const metrics = require('next-metrics');
+const gzip = denodeify(require('zlib').gzip);
+
+const concatenatedStylesCache = {};
+const concatenatedStylesSizeCache = {};
+let stylesheets;
+
+const calculateSize = content => {
+	return gzip(content)
+		.then(gzipped => ({
+			raw: Buffer.byteLength(content),
+			gzip: Buffer.byteLength(gzipped)
+		}))
+}
 
 module.exports = {
-	getStylesheets: (options, directory) => {
+
+	concatenateStyles: stylesheetNames => {
+		const hash = stylesheetNames.join(':');
+		if (!concatenatedStylesCache[hash]) {
+			concatenatedStylesCache[hash] = stylesheetNames.reduce((str, name) => {
+				/* istanbul ignore next */
+				if (!stylesheets[name]) {
+					throw new Error(`Stylesheet ${name}.css does not exist`);
+				}
+				return str + stylesheets[name];
+			}, '');
+			concatenatedStylesSizeCache[hash] = calculateSize(concatenatedStylesCache[hash])
+		}
+
+		concatenatedStylesSizeCache[hash]
+			.then(({raw, gzip}) => {
+				metrics.histogram('head_css_size.raw', raw);
+				metrics.histogram(`head_css_size.raw.${hash}`, raw);
+				metrics.histogram('head_css_size.gzip', gzip);
+				metrics.histogram(`head_css_size.gzip.${hash}`, gzip);
+			})
+
+
+		return concatenatedStylesCache[hash];
+	},
+
+	init: (options, directory) => {
 		const headCsses = fs.readdirSync(`${directory}/public`)
```

Comment:
Is this no longer _head_ cssess then? That filter would also pick up non-head CSSes?